pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/RealtimeSegmentConfig.java

```diff
@@ -68,7 +69,8 @@ private RealtimeSegmentConfig(String tableNameWithType, String segmentName, Stri
       RealtimeSegmentZKMetadata realtimeSegmentZKMetadata, boolean offHeap, PinotDataBufferMemoryManager memoryManager,
       RealtimeSegmentStatsHistory statsHistory, String partitionColumn, PartitionFunction partitionFunction,
       int partitionId, boolean aggregateMetrics, boolean nullHandlingEnabled, String consumerDir,
-      UpsertConfig.Mode upsertMode, PartitionUpsertMetadataManager partitionUpsertMetadataManager) {
+      UpsertConfig.Mode upsertMode, PartitionUpsertMetadataManager partitionUpsertMetadataManager,
+      String upsertComparisonColumn) {
```

Comment:
(nit) keep the same order as the declaration
```suggestion
      UpsertConfig.Mode upsertMode,
      String upsertComparisonColumn, PartitionUpsertMetadataManager partitionUpsertMetadataManager) {
```